### PR TITLE
urls don't have option to pass default protocol and blob endpoint url

### LINKED ISSUE #22: urls don't have option to pass default protocol and blob endpoint url
Looking for a way to contribute this feature but seems like there is permission issue.

I am looking to support urls like this which can have DefaultEndpointsProtocol and  BlobEndpoint mentioned.
url = "DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://127.0.0.1:10000/devstoreaccount1"

### PR DESCRIPTION
**Pull request by adityajaroli**

### CODE REVIEW

File: storefact/_urls.py [hunk 1]
```diff
@@ -44,6 +44,12 @@ def url2dict(url, raise_on_extra_params=False):
     if u'create_if_missing' in parsed['query']:
         create_if_missing = parsed['query'].pop(u'create_if_missing')[-1]  # use last appearance of key
         params['create_if_missing'] = create_if_missing in TRUEVALUES
+    if u'blob_endpoint' in parsed['query']:
```
Comment (sebastianneubauer) [inline storefact/_urls.py]: This looks like it can easily be buggy. I would suggest to refactor this out into a function with a good name (what is this even doing?) and add some unit tests of this function with some edge cases and some cases to show what this is expected to do.

Comment (sebastianneubauer) [reply storefact/_urls.py]: So, url2dict cound be something like:
```
def url2dict(url, raise_on_extra_params=False):
    ...
    params['create_if_missing'] = extract_create_if_missing(parsed)
    params['default_endpoints_protocol'] = extract_default_protocol(parsed)
    ...
```
and all of these extract_ functions can be unit tested properly (in the best case also without needing to mack anything and not needing to setup stuff in the test which is not needed for this function, e.g. create_if_missing is not needed in "extract_default_protocol").

**Review by sebastianneubauer — APPROVED** [review]

File: storefact/_urls.py (hunk 1 above)
Comment (adityajaroli) [reply storefact/_urls.py]: Makes sense. Let me implement it.

Comment (adityajaroli) [reply storefact/_urls.py]: Implemented review comments. Please review again.

Comment (sebastianneubauer) [reply storefact/_urls.py]: Great, lgtm! Good to go!
